dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java

```diff
@@ -687,6 +685,27 @@ private void validateUserRoles(User user, User currentUser, List<ErrorReport> er
     }
   }
 
+  @Override
+  public List<ErrorReport> validateUserRoleCreateOrUpdate(UserRole role, User currentUser) {
+
+    List<ErrorReport> errors = new ArrayList<>();
+
+    if (currentUser == null || role == null) {
+      return errors;
+    }
+
+    if (!currentUser.isSuper() && role.isSuper()) {
+      errors.add(new ErrorReport(UserRole.class, ErrorCode.E3032, currentUser.getUsername()));
+    }
+
+    UserRole userRoleBefore = userRoleStore.get(role.getId());
```

Comment:
i don't follow why we go to the `userRoleStore` to get the same `userRole` that we already seem to have. Is there a specific reason for this?

Reply:
This is done to compare with the original unmodified role, the incoming role could be mutated.